plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java

```diff
@@ -268,15 +268,16 @@ public ResultSet executeQuery()
         };
     }
 
+    @Override
+    public boolean supportsLimit(long limit)
+    {
+        return limit <= Integer.MAX_VALUE;
+    }
+
     @Override
     protected Optional<BiFunction<String, Long, String>> limitFunction()
     {
-        return Optional.of((sql, limit) -> {
-            if (limit > Integer.MAX_VALUE) {
-                return sql;
-            }
-            return sql + " LIMIT " + limit;
-        });
+        return Optional.of((sql, limit) -> sql + " LIMIT " + limit);
```

Comment:
can't you change `isLimitGuaranteed` to `true` now?

Reply:
`isLimitGuaranteed` changes limit to partial limit in plan. Does Phoenix return exactly 10 rows with LIMIT 10 or does it not?
I don't have context to know if it's safe to do so.

Reply:
I don't have context, but `isLimitGuaranteed` was false in Phoenix also before my recent changes.